### ISSUE TITLE
auth_base.js bug

### ISSUE THREAD

**Issue by celesteking** [baudehlo/Haraka#1107]
Essentially:

```
C: EHLO xxx
C: AUTH LOGIN
C: AUTH LOGIN
bug triggered
```

**Comment by celesteking** [comment]
my proposal
```diff
--- a/plugins/auth/auth_base.js
+++ b/plugins/auth/auth_base.js
@@ -182,6 +182,9 @@ exports.auth_login = function(next, connection, params) {
         ( connection.notes.auth_login_asked_login &&
          !connection.notes.auth_login_userlogin))
     {
+        if (!params[0])
+            return next(DENYDISCONNECT, 'bad protocol');
+
         var login = utils.unbase64(params[0]);
         connection.respond(334, LOGIN_STRING2, function () {
             connection.notes.auth_login_userlogin = login;
```

**Comment by baudehlo** [comment]
+1 - please submit a PR.

**Comment by Dexus** [comment]
can be closed as merged #1121
